stacks/BasicApiStack.ts

```diff
@@ -2,12 +2,45 @@ import { StackContext, Api, Table, Queue, Bucket, Topic, Config } from '@serverl
 import { SSTConstruct } from '@serverless-stack/resources/dist/Construct'
 import * as cfnApig from 'aws-cdk-lib/aws-apigatewayv2'
 import * as apig from '@aws-cdk/aws-apigatewayv2-alpha'
+import { Duration } from 'aws-cdk-lib'
 
 export function BasicApiStack ({
   app,
   stack
 }: StackContext): { queue: Queue, bucket: Bucket, dynamoDbTable: Table } {
-  const dlq = new Queue(stack, 'PinDlq')
+  const dynamoDbTable = new Table(stack, 'BasicV2', {
+    fields: {
+      cid: 'string'
+    },
+    primaryIndex: {
+      partitionKey: 'cid'
+    }
+  })
+
+  const dlq = new Queue(stack, 'PinDlq', {
+    cdk: {
+      queue: {
+        visibilityTimeout: Duration.seconds(40) // should be greater than the lambda timeout
+      }
+    },
+    consumer: {
+      function: {
+        timeout: '30 seconds',
```

Comment:
lambda will timeout and fail the batch if it takes more than 30s. then 10s later the messages it was trying to handle will be returned to the queue (visibilityTimeout: 40s)